src/main/java/org/jacp/controller/CreateCompetitionController.java

```diff
@@ -0,0 +1,61 @@
+package org.jacp.controller;
+
+import org.jacp.dto.CompetitionDto;
+import org.jacp.dto.ParticipantsDto;
+import org.jacp.dto.QuestionDto;
+import org.jacp.dto.SearchDto;
+import org.jacp.entity.CompetitionEntity;
+import org.jacp.enums.Status;
+import org.jacp.mapper.CompetitionMapper;
+import org.jacp.service.CompetitionService;
+import org.jacp.service.QuestionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+/**
+ * @author saffchen created on 04.12.2023
+ */
+@RestController
+@RequestMapping(value = CreateCompetitionController.URL, produces = MediaType.APPLICATION_JSON_VALUE)
+public class CreateCompetitionController {
+
+    static final String URL = "/api/v1/competition";
+    @Autowired
+    private CompetitionMapper mapper;
+    @Autowired
+    private QuestionService questionService;
+    @Autowired
+    private CompetitionService competitionService;
+
+    @PostMapping("/create")
+    public ResponseEntity<CompetitionEntity> createCompetition(@RequestBody SearchDto searchDto) {
+        List<QuestionDto> questions = questionService.responseFromQuestionServiceFilter(searchDto).getBody();
```

Comment:
Minor: let's rename method to exclude "question service" words, it could be like "getQuestionsByFillter()". But if requires filter, then it's better to map the SearchDto to FilterDto in advance.

Reply:
Fixed